src/ethereum/frontier/state.py

```diff
@@ -35,6 +35,53 @@ class State:
         default_factory=lambda: Trie(secured=True, default=None)
     )
     _storage_tries: Dict[Address, Trie[U256]] = field(default_factory=dict)
+    _snapshots: List[
+        Tuple[Trie[Optional[Account]], Dict[Bytes, Trie[U256]]]
+    ] = field(default_factory=list)
+
+
+def start_transaction(state: State) -> None:
+    """
+    Start a state transaction. Transactions are entirely implicit and can be
+    nested. It is not possible to calculate the state root during a
+    transaction.
+
+    Parameters
+    ----------
+    state : State
+        The state.
+    """
+    state._snapshots.append(
+        (
+            copy_trie(state._main_trie),
+            {k: copy_trie(t) for (k, t) in state._storage_tries.items()},
+        )
+    )
+
+
+def commit_transaction(state: State) -> None:
+    """
+    Commit a state transaction.
+
+    Parameters
+    ----------
+    state : State
+        The state.
+    """
+    state._snapshots.pop()
+
+
+def abort_transaction(state: State) -> None:
```

Comment:
I prefer the name `rollback_transaction`. Thoughts?

Reply:
Agreed.